server/src/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1705,6 +1709,14 @@ public VMTemplateVO createPrivateTemplateRecord(CreateTemplateCmd cmd, Account t
                 s_logger.debug("This template is getting created from other template, setting source template Id to: " + sourceTemplateId);
             }
         }
+
+
+        // ignore passed zoneId if we are using region wide image store
+        List<ImageStoreVO> stores = _imgStoreDao.findRegionImageStores();
+        if (stores != null && stores.size() > 0) {
```

Comment:
Hello @syed

this if condition can be changed to use the Apache's CollectionUtils  isEmpty() method for enhanced readability https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html#isEmpty(java.util.Collection)